docs/language/contracts.rst

```diff
@@ -125,6 +138,25 @@ The sequence of the accounts in the ``AccountMeta`` array matters and must follo
 :ref:`IDL ordering <account_management>`.
 
 
+.. _solana_contract_call:
+
+Calling a contract on Solana
+____________________________
+
+On Solana, contracts are deployed to a program account, which holds only the contract's executable binary. For this
```

Comment:
"On Solana, contracts are deployed to a program account, which holds only the contract's executable binary. "

This is not clear. Where does the data go then?

Reply:
The explanation about the data account is on line 108. I did not think it was necessary to repeat it here, because data account are not necessary for calling a contract (from now on, they are not necessary if we do not modify any contract variable, regardless of the function mutability). Perhaps we should add a section explaining how contracts are laid out on Solana, so we do not need to repeat the inner principle everywhere. What do you think?

Reply:
This whole paragraph can be written better. The heading is "Calling a contract on Solana". First explain how to make a call on Solana, and then possibly give some background information.

https://solang.readthedocs.io/en/v0.3.2/language/types.html#contract-types needs updating to reflect that contract types are not allowed.

Reply:
> https://solang.readthedocs.io/en/v0.3.2/language/types.html#contract-types needs updating to reflect that contract types are not allowed.

Good point. I had forgotten about that.